src/Microsoft.SqlTools.ServiceLayer/QueryExecution/DataStorage/SaveAsExcelFileStreamWriterHelper.cs

```diff
@@ -0,0 +1,688 @@
+﻿
```

Comment:
Yep, I consulted with our legal folks real quick to make sure I fully understood our CLA. You will still own the copyright of the code you contribute to our project. You would be able to use your code however you like, without any need to attribute Microsoft. So to sum it up, yes, you can totally reuse your code to make a general export to Excel library.